src/Network/Response.php

```diff
@@ -1665,4 +1707,307 @@ public function stop($status = 0)
     {
         exit($status);
     }
+
+    /**
+     * Retrieves the HTTP protocol version as a string.
+     * The string MUST contain only the HTTP version number (e.g., "1.1", "1.0").
+     *
+     * @return string HTTP protocol version.
+     */
+    public function getProtocolVersion()
+    {
+        return preg_replace('/[^0-9.]/', '', $this->_protocol);
+    }
+
+    /**
+     * Return an instance with the specified HTTP protocol version.
+     *
+     * The version string MUST contain only the HTTP version number (e.g.,
+     * "1.1", "1.0").
+     *
+     * This method MUST be implemented in such a way as to retain the
+     * immutability of the message, and MUST return an instance that has the
+     * new protocol version.
+     *
+     * @param string $version HTTP protocol version
+     * @return static
+     */
+    public function withProtocolVersion($version)
+    {
+        if (!in_array($version, ['1.0', '1.1', '2.0'])) {
+            throw new InvalidArgumentException(sprintf('Invalid HTTP protocol version: %s', $version));
+        }
+
+        $tmp = $this->_protocol;
+        $this->_protocol = $version;
+        $clone = clone $this;
+        $this->_protocol = $tmp;
```

Comment:
You should be modifying `$clone` and never `$this` in the `withXX` methods.

```php
$clone = clone $this;
$clone->_protocol = $version;

return $clone
```

This will work even if the properties are private because PHP visibility is class level not object level.